### ISSUE TITLE
[Rules] False positive regarding usage of 'aria-valuenow' on role of 'separator' #3816

### ISSUE THREAD

**Issue by aliunwala** [IBMa/equal-access#923]
Testcase: https://codesandbox.io/s/issue-3816-byogx?file=/index.html

The div[role=separator] is triggering a violation indicating that usage of `aria-valuenow` is invalid. However, `aria-valuenow` is valid if the separator is focusable - and it is in the testcase.

See https://www.w3.org/TR/wai-aria-1.2/#separator

Note also that role=separator isn't detected as a 'widget role'

Reported by Steven Herlihy (https://ibmaustincomms.slack.com/archives/C036P1CTN/p1637141944154400)

![image](https://user-images.githubusercontent.com/4451128/171962966-620593bb-257c-4d82-b8e2-10f7d0a3db4e.png)
